tests/test_api.rs

```
@@ -1553,6 +1563,29 @@ fn object() {
     assert!(!object.has(scope, n_unused).unwrap());
     assert!(object.delete(scope, n1.into()).unwrap());
     assert!(!object.has(scope, n1.into()).unwrap());
+
+    let global = context.global(scope);
+    let object_string = v8::String::new(scope, "object").unwrap().into();
+    global.set(scope, object_string, object.into());
+    assert!(object
+      .set_integrity_level(scope, v8::IntegrityLevel::Sealed)
+      .unwrap());
+    {
+      let scope = &mut v8::TryCatch::new(scope);
+      // Creating a new property fails (in strict mode).
+      eval_strict(scope, "object.c = 'foo';");
+      assert!(scope.has_caught());
+    }
+
+    assert!(object
+      .set_integrity_level(scope, v8::IntegrityLevel::Frozen)
+      .unwrap());
+    {
+      let scope = &mut v8::TryCatch::new(scope);
+      // Writing an existing property fails (in strict mode).
+      eval_strict(scope, "object.a = 'foo';");
+      assert!(scope.has_caught());
```

In non-strict mode, this silently does nothing.